File: Windows/Nim/x64/bsod.nim
proc RtlAdjustPrivilege(Privilege: int32, Enable: bool, CurrentThread: bool, Enabled: ptr bool): int32 {.stdcall, dynlib: "ntdll.dll", importc.}
proc NtRaiseHardError(ErrorStatus: uint32, NumberOfParameters: uint32, UnicodeStringParameterMask: uint32, Parameters: pointer, ValidResponseOptions: uint32, Response: ptr uint32): int32 {.stdcall, dynlib: "ntdll.dll", importc.}

var enabled: bool
var response: uint32

discard RtlAdjustPrivilege(19, true, false, addr enabled)
discard NtRaiseHardError(0xC0000420'u32, 0, 0, nil, 6, addr response)

File: Windows/Delphi/x32/bsod.pas
program bsod;

{$mode objfpc}

function RtlAdjustPrivilege(Privilege: LongInt; Enable: Boolean; CurrentThread: Boolean; var Enabled: Boolean): LongInt; stdcall; external 'ntdll.dll';
function NtRaiseHardError(ErrorStatus: LongWord; NumberOfParameters: LongWord; UnicodeStringParameterMask: LongWord; Parameters: Pointer; ValidResponseOptions: LongWord; var Response: LongWord): LongInt; stdcall; external 'ntdll.dll';

var
  enabled: Boolean;
  response: LongWord;

begin
  RtlAdjustPrivilege(19, True, False, enabled);
  NtRaiseHardError($C0000420, 0, 0, nil, 6, response);
end.

